src/librustdoc/clean/types.rs

```diff
@@ -586,9 +586,10 @@ impl Item {
     }
     crate fn has_stripped_fields(&self) -> Option<bool> {
         match *self.kind {
-            StructItem(ref _struct) => Some(_struct.fields_stripped),
-            UnionItem(ref union) => Some(union.fields_stripped),
-            VariantItem(Variant::Struct(ref vstruct)) => Some(vstruct.fields_stripped),
+            StructItem(ref _struct) => Some(_struct.has_stripped_fields()),
+            UnionItem(ref _union) => Some(_union.has_stripped_fields()),
+            EnumItem(ref _enum) => Some(_enum.has_stripped_fields()),
+            VariantItem(ref v) => v.has_stripped_fields(),
```

Comment:
Why did the patterns change? Also, can you rename e.g. `_struct` to `struct_` so we get dead code analysis?

Reply:
Because we also handle `EnumItem` in here to have one unified call.

I'll rename too.